src/main/java/com/dataloom/organization/OrganizationsApi.java

```diff
@@ -1,5 +1,152 @@
+    Void removeAutoApprovedEmailDomains( @Path( ID ) UUID organizationId, @Body Set<String> emailDomain );
+
+    /**
+     * Adds a single e-mail domain to the auto-approval list. This will fail for users who are not an owner of the
+     * organization.
+     *
+     * @param organizationId The id of the organization to modify.
+     * @param emailDomain    The e-mail domain to add to the auto-approval list.
+     */
+    @PUT( ORGANIZATIONS + ID_PATH + EMAIL_DOMAINS + EMAIL_DOMAIN_PATH )
+    Void addAutoApprovedEmailDomain( @Path( ID ) UUID organizationId, @Path( EMAIL_DOMAIN ) String emailDomain );
+
+    /**
+     * Removes a single e-mail domain to the auto-approval list. This will fail for users who are not an owner of the
+     * organization.
+     *
+     * @param organizationId The id of the organization to modify.
+     * @param emailDomain    The e-mail domain to add to the auto-approval list.
+     */
+    @DELETE( ORGANIZATIONS + ID_PATH + EMAIL_DOMAINS + EMAIL_DOMAIN_PATH )
+    Void removeAutoApprovedEmailDomain( @Path( ID ) UUID organizationId, @Path( EMAIL_DOMAIN ) String emailDomain );
+
+    @GET( ORGANIZATIONS + ID_PATH + PRINCIPALS )
+    Set<Principal> getPrincipals( @Path( ID ) UUID organizationId );
+
+    /**
+     * This is a convenience call that modifies that grants {@code {@link com.dataloom.authorization.Permission#READ}}
+     * on the specified organization to the specified principals. A user's membership in an organization is determined
+     * by whether or not they have {@code {@link com.dataloom.authorization.Permission#READ}} on the organization.
+     *
+     * @param organizationId The id of the organization.
+     * @param principals     A set of valid principals.
+     */
+    @POST( ORGANIZATIONS + ID_PATH + PRINCIPALS )
+    Void addPrincipals( @Path( ID ) UUID organizationId, @Body Set<Principal> principals );
+
+    /**
+     * This is a convenience call that modifies that grants {@code {@link com.dataloom.authorization.Permission#READ}}
+     * on the specified organization to the specified principals. A user's membership in an organization is determined
+     * by whether or not they have {@code {@link com.dataloom.authorization.Permission#READ}} on the organization.
+     *
+     * @param organizationId The id of the organization.
+     * @param principals     A set of valid principals.
+     */
+    @PUT( ORGANIZATIONS + ID_PATH + PRINCIPALS )
+    Void addPrincipals( @Path( ID ) UUID organizationId, @Body Set<Principal> principals );
+
+    /**
+     * This is a convenience call that modifies that removes all {@code {@link com.dataloom.authorization.Permission}}s
+     * that a principal may have on an organization, including discover.
+     *
+     * @param organizationId The id of the organization.
+     * @param principals     A set of valid principals
+     */
+    @HTTP( method = "DELETE", hasBody = true, path = ORGANIZATIONS + ID_PATH + PRINCIPALS )
```

Comment:
this is different... why `@HTTP` with `hasBody` and `path` instead of `@DELETE`?